### PR TITLE
Replace overview when site is with a pending migration

### LINKED ISSUE #95379: Post-purchase Migration Experience: Handle selecting a DIFM site in the Sites dashboard that is pending migration
## Description
When a user clicks on a site in the Sites dashboard that has the `migration-pending-difm` sticker, display a detailed view communicating that the site is ready to be migrated.

Clicking on the _Start your migration_ button goes to the _Tell us about your WordPress site_ (i.e. credentials) step of the migration flow. We should ensure that this button click can be analyzed in Tracks funnels so that we can discern that the user entered the flow from the Sites dashboard.

Figma - nOWSHCjXMpB5sXHthTVdvK-fi-3949_19810

Image

### LINKED ISSUE #95380: Post-purchase Migration Experience: Handle selecting a site in the Sites dashboard that is pending migration
Dependent on #95379.

## Description
When a user clicks on a site in the Sites dashboard that has the `migration-pending` or `migration-pending-diy` sticker, display a detailed view communicating that the site is ready to be migrated.

Clicking on the _Start your migration_ button goes to the:
- `migration-pending` - _How do you want to migrate?_ step of the migration flow.
- `migration-pending-diy` - _Let's migrate your site_ (i.e. migration instructions) step of the migration flow.

In either case, we should ensure that the button click can be analyzed in Tracks funnels so that we can discern that the user entered the flow from the Sites dashboard.

Figma - nOWSHCjXMpB5sXHthTVdvK-fi-3949_19810

Image

### PR DESCRIPTION
**Pull request by renatho**
Related to #95379 and #95380

## Proposed Changes

* Extract some components from the "Hosting Features" tab, so we can reuse it.
* It customizes the Sites overview with a button to continue the migration when it's pending.
* The button contains a `ref` as `hosting-migration-overview`, so we can identify when the user is coming from this page.

#### This PR Doesn't Include

* Disable other tabs from the menu.
* Update the header when navigating to the site, similar to the changes we have in the listing.

@donnapep, do you think it's fine to deploy it as it is and implement these other parts in other PRs? I think one issue would be fine for both changes. I'll wait for your answer to then create the new issue, otherwise, I'll continue working on this one.

## Why are these changes being made?

* We want to offer a way for the users to continue their migration if they stopped in the middle.

## Testing Instructions

* If https://github.com/Automattic/wp-calypso/pull/95413 wasn't merged yet, merge it into this branch to make tests more real and easier.
* Navigate to the flow `/setup/migration`.
* Navigate until the "How to migrate" step.
* Navigate to `/sites`.
* Click on the site, and click on "Start your migration".
* Make sure you navigate back to where you were in the flow.
* Click on the "Do it for me" option.
* Navigate again to the `/sites` and click on "Start your migration".
* Make sure you navigate back to your choice.
* Return one step and choose the "Do it myself".
* Navigate to the `/sites`.
* Check that you won't see the button because it changes quickly to the "started" status that will be implemented as part of https://github.com/Automattic/wp-calypso/issues/95383.
* Now repeat the tests using the flow `/setup/hosted-site-migration` and make sure it works properly.

## Screenshots

Screenshot 2024-10-18 at 18 40 00

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/hosting/hosting-features/components/hosting-features.tsx
Comment (renatho) [inline client/hosting/hosting-features/components/hosting-features.tsx:31]: Components were extracted so that we could reuse them.

File: client/hosting/hosting-features/components/hosting-features.tsx
Comment (renatho) [inline client/hosting/hosting-features/components/hosting-features.tsx:254]: Components were extracted so that we could reuse them.

**Comment by renatho** [comment]
> @donnapep, do you think it's fine to deploy it as it is and implement these other parts in other PRs? I think one issue would be fine for both changes. I'll wait for your answer to then create the new issue, otherwise, I'll continue working on this one.

I added the "[Status] String Freeze" because we have many new texts, so we'll delay a little to merge this anyway.

File: client/hosting/sites/components/dotcom-style.scss
Comment (renatho) [inline client/hosting/sites/components/dotcom-style.scss:482]: This fixes the spacing at the bottom of the tabs.

Without this, it wasn't respecting the padding.

Screenshot 2024-10-18 at 18 32 26

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/16927618

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @renatho for including a screenshot in the description! This is really helpful for our translators.

File: client/sites-dashboard/test/utils.js
Comment (m1r0) [inline client/sites-dashboard/test/utils.js]: The `isMigrationInProgress` import can be added here as well.

**Review by m1r0 — APPROVED** [review]
Looks good and works as described! Thanks for splitting those components.

I've added one small non-blocker suggestion.

**Comment by donnapep** [comment]
@renatho Could you add some additional context or screen shots of what these mean? It's a bit unclear to me.

> Disable other tabs from the menu.
> Update the header when navigating to the site, similar to the changes we have in the listing.

**Comment by renatho** [comment]
> @renatho Could you add some additional context or screen shots of what these mean? It's a bit unclear to me.

Sure! I was referring to these parts:

Screenshot 2024-10-21 at 14 09 37

File: client/sites-dashboard/test/utils.js
Comment (renatho) [reply client/sites-dashboard/test/utils.js]: Ah! Good catch! Fixed: https://github.com/Automattic/wp-calypso/pull/95497/commits/81f3cee6b410f4263012173f24091b06e961cf33

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/16927618).

**Comment by donnapep** [comment]
@renatho Thank, I missed those subtleties. I can create an issue for adding the status to the header.

@fditrapani Can you confirm that we should disable the _Hosting settings_ tab for pending / started migrations?

**Comment by fditrapani** [comment]
Thanks for the ping. This looks good from the screenshots. We had some back and forth about this with Michael. I figured it made sense to keep it disabled so you can stay focused on starting your migration. I can see the other side of it that people might want to poke around so I can be convinced otherwise. What do you all think?

File: client/hosting/overview/components/migration-overview.tsx
Comment (Imran92) [inline client/hosting/overview/components/migration-overview.tsx]: Just a little concern, as we're only checking if the site is on a free plan or not, in case the site is on a `Personal` or `Premium` plan, will it send the user to the credentials or how-to-migrate step instead of the pre-checkout step?

Comment (renatho) [reply client/hosting/overview/components/migration-overview.tsx]: Hmm!! Good point, @Imran92! I was thinking that in our flows users wouldn't be able to pick a Personal or Premium plan and then try to migrate, but after you said, I tested it and I could do it through the `/start`, buying the plan (Personal or Premium), select the intent to migrate and use the flow.

I'll update it to a different check.

Comment (renatho) [reply client/hosting/overview/components/migration-overview.tsx]: Updated the check to match what we check in the flows to see if we'll redirect the user to the upgrade plan step or not. Now it makes more sense! Thank you for reviewing and catching this!

I also took the opportunity to refactor the duplicate code.

https://github.com/Automattic/wp-calypso/pull/95497/commits/3887d73cf58abd97025a2d745e6564fa5cc81d6d https://github.com/Automattic/wp-calypso/pull/95497/commits/2da505cadcab85fcba57cd756599270f2a7d7a18

Comment (m1r0) [reply client/hosting/overview/components/migration-overview.tsx]: @renatho can you update the unit tests as well?

Comment (renatho) [reply client/hosting/overview/components/migration-overview.tsx]: Oops! Fixed here: https://github.com/Automattic/wp-calypso/pull/95497/commits/e5d856e258bcbb497c36a5665d59b4802c6261a9

**Comment by renatho** [comment]
@m1r0 and @Imran92, since we had some iterations after the approval, I'll wait for a new approval, okay?

File: client/hosting/overview/components/migration-overview.tsx
Comment (Imran92) [inline client/hosting/overview/components/migration-overview.tsx]: According to the [issue](https://github.com/Automattic/wp-calypso/issues/95380), in case of `migration-pending-diy` sticker, the user should be taken to the `site-migration-instructions` step. Looks like this here is taking us to the `migration-how-to-migrate` step. In case it's not intentional, should we fix it?

image

Comment (renatho) [reply client/hosting/overview/components/migration-overview.tsx]: Good catch, @Imran92! Fixed in https://github.com/Automattic/wp-calypso/pull/95497/commits/2de7505061f8eca40813a2ca787f30d34491d2e3

**Review by Imran92 — COMMENTED** [review]
I'm seeing a weird issue, but not sure why it's happening.

I had a site in a pending state. In the overview, section worked as expected, showing me correct button and everything relevant.

Then I changed the sticker to 'migration-completed-*'. But even after refreshing and hard-refreshing the page section many times, I still see the the migration-pending overview page.

https://github.com/user-attachments/assets/daa5ae91-ce05-474a-ae5a-cde82288df2c

But apart from that, LGTM. So I'm approving. Feel free to look into the issue either separately or here.

**Review by Imran92 — APPROVED** [review]

**Comment by renatho** [comment]
@Imran92, there's a cache, but after 1 or 2 minutes it's updated. Could you confirm if it works for you after 1 or 2 minutes? I tested a few times and it's working well for me, just with this small cache.

**Comment by renatho** [comment]
I just removed a not-needed conditional that I left accidentally: https://github.com/Automattic/wp-calypso/pull/95497/commits/456d3aa071b4c71b6057595fce550aca3bc1eb87

**Comment by renatho** [comment]
Sorry, but I did 2 last minute changes that I noticed while working on another PR:

1. I removed a not-needed conditional that I left accidentally: https://github.com/Automattic/wp-calypso/commit/456d3aa071b4c71b6057595fce550aca3bc1eb87. It wasn't being used at all since we are checking this in the parent component.
2. I also changed the tests to also mock another dependency, similar to the others: https://github.com/Automattic/wp-calypso/pull/95497/commits/84a44ecc89d0520aeb420164b84bf9586b661975

**Review by Imran92 — APPROVED** [review]
LGTM 👍 Cache gets cleared after a couple of minutes

**Comment by renatho** [comment]
I just rebased to fix a conflict, so I'm doing a sanity test, and I'll wait for the tests and merge it.

**Pull request merged**
